### ISSUE TITLE
Fine tune application's dialogs for IMU Calibration and IK

### ISSUE THREAD

**Issue by aymanhab** [opensim-org/opensim-gui#1236]
### Actual result

(include screen grabs where appropriate)

**Comment by aymanhab** [comment]
Items per feedback from @jenhicks 
1. There is not a good mapping between (1) xml file name and the properties in the xml file and (2) the names in the GUI tools (e.g., calibration vs placer ... align vs. heading correction). We should use consistent terminology where possible. 

2. It's not completely clear to users from the set-up how to skip the heading correction step. ... maybe note that it is optional?

3. The xml set-up file lists rotation in radians, but the GUI tool asks for degrees. 

4. Is there a reason to have the model name in xml and the model filename be different?

5. We could consider removing markers from the model file, but I'm open to keeping them.

6. The name of the set-up file and quaternions file is quite long and it's not completely self-evident what it is. What is the rationale for the hip rom task over a walking portion?

7. If I hit the "Weights" button before loading the set-up file, I get an error (see end of message). Should handle this more gracefully.

8. When I am running IK, the GUI says in the bottom right "Executing IMU Model Calibration."

**Comment by carmichaelong** [comment]
> There is not a good mapping between (1) xml file name and the properties in the xml file and (2) the names in the GUI tools (e.g., calibration vs placer ... align vs. heading correction). We should use consistent terminology where possible.

A few suggestions to get started (definitely feel free to work from these and refine):
- imuPlacer.xml -> Setup_IMU_Calibration.xml [so that "calibration" is clear, rather than "placer"]
- I'm not sure I would understand "default pose" in the "Orientation file at default pose". Perhaps "registration pose"? "Calibration pose" would make the "calibration" terminology consistent with the xml file, but I don't know if that's the right term there?
- "Align sensor (name, axis)" -> "Align base IMU sensor (name, axis)"
- Description of "sensor_to_opensim_rotations" could be the same with the GUI description too. Perhaps: "Space fixed Euler angles (XYZ order) from IMU space to OpenSim space" (this would be for both calibration and IMU IK)
- In IMU IK: the setup file has "results_directory" but the GUI has "Motion File"
- A general comment: We use both "IMU" and "sensor" to refer to the physical devices. I don't think it's too confusing that we use both references, but if others thing it'd be clearer to pick one and use it, it might be useful.

> It's not completely clear to users from the set-up how to skip the heading correction step. ... maybe note that it is optional?

- Agreed. Perhaps give this section a checkbox to enable/disable to make it super clear that it's optional? If the checkbox's label had a description like "Perform heading correction", this could also help make it clear what this selection is for.
- I'm not sure if this is possible, but one step further could even be helpful: make it so the checkbox cannot be changed until an orientation file is given (since the drop-downs can't populate until the orientations file is given)

> The name of the set-up file and quaternions file is quite long and it's not completely self-evident what it is. What is the rationale for the hip rom task over a walking portion?

- Could use same data as the current OpenSense example on Confluence?

**Comment by aymanhab** [comment]
> > There is not a good mapping between (1) xml file name and the properties in the xml file and (2) the names in the GUI tools (e.g., calibration vs placer ... align vs. heading correction). We should use consistent terminology where possible.
> 
> A few suggestions to get started (definitely feel free to work from these and refine):
> 
> * imuPlacer.xml -> Setup_IMU_Calibration.xml [so that "calibration" is clear, rather than "placer"]
I agree this would be helpful but my understanding is that the issue is not the xml file name but rather the discrepancy between the menu pick name (IMU Calibrate...)/IMU Calibration Tool on one side and the class name (IMUPlacer). The general idea is that IMUPlacer is one way to calibrate and others may come later, maybe replace Calibrate/Calibration with Place/Placement or Registration?

> * I'm not sure I would understand "default pose" in the "Orientation file at default pose". Perhaps "registration pose"? "Calibration pose" would make the "calibration" terminology consistent with the xml file, but I don't know if that's the right term there?
If we introduce Registration above then yes if not then we're adding another term to the confusion and I'd lean to use Calibration or Placement.

> * "Align sensor (name, axis)" -> "Align base IMU sensor (name, axis)"

OK though too wordy, should we keep sensor? 

> * Description of "sensor_to_opensim_rotations" could be the same with the GUI description too. Perhaps: "Space fixed Euler angles (XYZ order) from IMU space to OpenSim space" (this would be for both calibration and IMU IK)

Sure
> * In IMU IK: the setup file has "results_directory" but the GUI has "Motion File"

This was a compromise since we expose only output file for GUI similar to marker based but the tool can produce other files e.g. errors etc. 
> * A general comment: We use both "IMU" and "sensor" to refer to the physical devices. I don't think it's too confusing that we use both references, but if others thing it'd be clearer to pick one and use it, it might be useful.
> 
> > It's not completely clear to users from the set-up how to skip the heading correction step. ... maybe note that it is optional?
> 
> * Agreed. Perhaps give this section a checkbox to enable/disable to make it super clear that it's optional? If the checkbox's label had a description like "Perform heading correction", this could also help make it clear what this selection is for.
> * I'm not sure if this is possible, but one step further could even be helpful: make it so the checkbox cannot be changed until an orientation file is given (since the drop-downs can't populate until the orientations file is given)
> 
> > The name of the set-up file and quaternions file is quite long and it's not completely self-evident what it is. What is the rationale for the hip rom task over a walking portion?

Sure open to shorter more descriptive name, this was a strawman to get the example in place. 
> 
> * Could use same data as the current OpenSense example on Confluence?

**Comment by carmichaelong** [comment]
@aymanhab Thanks for the notes, some responses below:
- I see now what you mean about Calibration vs. Placer. I think maybe then switch all of those to "placer":
1. Drop down: "IMU Calibrate..." -> "IMU Placer..."
2. IMU Placer window title: "IMU Model Calibration Tool" -> "IMU Placer Tool" (I think this is OK until there are more Calibration options, which would then change this tool?)
3. IMU Placer subsection name: "Model Calibration" -> "IMU Placement"
4. "Orientation file at default pose" -> "Orientation file at IMU placement pose"
5. (sort of related): "imuPlacer.xml" -> "Setup_IMU_Placer.xml" or "IMUPlacer_Setup.xml" (or something like this. not too picky here, but I like having "setup" in the name to note that this can be loaded by a tool)
- "Align sensor (name, axis)" -> "Align base IMU (name, axis)" (I agree that "sensor" can be removed)
- Re "results_directory" vs "Motion File": I did a little testing and realized that if you open the provided setup file for IMU-IK and then save it, the tags are different. Suggested next steps:
1. I think this setup file should just be updated and much of the confusion is cleared up?
2. The updated .xml file has a slightly odd order of properties. The input "orientations_file" is at the bottom, and the "output_motion_file" is midway down the file. These could get swapped to make the flow a little better.
- Re hip joint trial as example data: Perhaps we could just replace this with some of the data from the OpenSense example on Confluence? Suggested steps:
1. Remove HJC trial, replace with walking data from OpenSense example
2. Shorten name of .xml setup file: "Setup_IMU-IK.xml"?
- Any thoughts on the suggestion above to make the heading correction step an optional section with a checkbox? Steps:
1. Perhaps give this section a checkbox to enable/disable to make it super clear that it's optional? If the checkbox's label had a description like "Perform heading correction", this could also help make it clear what this selection is for.
2. I'm not sure if this is possible, but one step further could even be helpful: make it so the checkbox cannot be changed until an orientation file is given (since the drop-downs can't populate until the orientations file is given)

**Comment by aymanhab** [comment]
Thanks @carmichaelong I'll go with Placer/Placement over Registration or Calibration since it parallels Marker Placement. @jenhicks please let us know if you have a different preference.

**Comment by aymanhab** [comment]
@jenhicks I'm making the changes per @carmichaelong suggestion please let me know if your preference for naming. Thank you

**Comment by aymanhab** [comment]
@carmichaelong In regards to the order of properties in the file, this is controlled by the inheritance (subclasses add properties to the end) as such results directory will appear near the top because it comes from parent class, the rest however can be arranged as needed.

**Comment by aymanhab** [comment]
The data files from the example are pretty long (600+ seconds) which doesn't add much value, will trim.

**Comment by aymanhab** [comment]
I believe this is done too, could be checked in latest build.
